internal/parse/parser.go

```diff
@@ -187,57 +214,108 @@ func isNameByte(c byte) bool {
 		'0' <= c && c <= '9' || c == '_'
 }
 
-type parseHelperTest struct {
-	bytef    func(byte) bool
-	stringf  func(string) bool
-	stringf0 func() bool
-	result   []bool
-	input    string
-	data     []string
+// skipName returns false if the parser is not on a name. Otherwise it advances
+// the parser until it is on the first non name byte and returns true.
+func (p *Parser) skipName() bool {
+	if p.pos >= len(p.input) {
+		return false
+	}
+	start := p.pos
+	for p.pos < len(p.input) && isNameByte(p.input[p.pos]) {
+		p.pos++
+	}
+	return p.pos > start
 }
 
-func TestRunTable(t *testing.T) {
-	var p = NewParser()
-	var parseTests = []parseHelperTest{
+// Functions with the prefix parse attempt to parse some construct. They return
+// the construct, and an error and/or a bool that indicates if the the construct
+// was successfully parsed.
+//
+// Return cases:
+//  - bool == true, err == nil
+//		The construct was sucessfully parsed
+//  - bool == false, err != nil
+//		The construct was recognised but was not correctly formatted
+//  - bool == false, err == nil
+//		The construct was not the one we are looking for
 
-		{bytef: p.peekByte, result: []bool{false}, input: "", data: []string{"a"}},
-		{bytef: p.peekByte, result: []bool{false}, input: "b", data: []string{"a"}},
-		{bytef: p.peekByte, result: []bool{true}, input: "a", data: []string{"a"}},
+// parseIdentifier parses either a name made up only of nameBytes or an
+// asterisk.
+func (p *Parser) parseIdentifier() (string, bool) {
+	if p.skipByte('*') {
+		return "*", true
+	}
 
-		{bytef: p.skipByte, result: []bool{false}, input: "", data: []string{"a"}},
-		{bytef: p.skipByte, result: []bool{false}, input: "abc", data: []string{"b"}},
-		{bytef: p.skipByte, result: []bool{true, true}, input: "abc", data: []string{"a", "b"}},
+	idStart := p.pos
+	if p.skipName() {
+		return p.input[idStart:p.pos], true
+	}
+	return "", false
+}
 
-		{bytef: p.skipByteFind, result: []bool{false}, input: "", data: []string{"a"}},
-		{bytef: p.skipByteFind, result: []bool{false, true, true}, input: "abcde", data: []string{"x", "b", "c"}},
-		{bytef: p.skipByteFind, result: []bool{true, false}, input: "abcde ", data: []string{" ", " "}},
+// parseFullName parses a column name or Go field name, optionally dot-prefixed by
+// its table or type name respectively.
+func (p *Parser) parseFullName(idc idClass) (FullName, bool) {
+	cp := p.save()
+	var fn FullName
+	p.skipSpaces()
```

Comment:
This is still skipping spaces. I thought we had agreed this was a job for the call site on the previous review / call?

Reply:
Ah maybe I changed the wrong version before. Its fixed now.